File: Data Structures & Algorithms/house-robber-ii/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if not nums:
            return 0
        if len(nums) == 1:
            return nums[0]
        if len(nums) == 2:
            return nums[0] if nums[0] > nums[1] else nums[1]
        
        return max(self.helper(nums[1:]), self.helper(nums[:-1]))

    def helper(self, nums):
        if not nums:
            return 0
        if len(nums) == 1:
            return nums[0]
        if len(nums) == 2:
            return nums[0] if nums[0] > nums[1] else nums[1]
        
        rob1 = rob2 = 0

        for n in nums:
            temp = max(n + rob1, rob2)
            rob1 = rob2
            rob2 = temp
        return rob2
        

File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        if (nums.length == 0) {
            return 0;
        }
        int k = nums.length;
        int startIndex = 0, endIndex = nums.length - 1;
        while (nums[endIndex] == val) {
            k--;
            endIndex--;
            if (endIndex <= startIndex) {
                return 0;
            }
        }
        while (startIndex < endIndex) {
            if (nums[startIndex] == val) {
                k--;
                int temp = nums[startIndex];
                nums[startIndex] = nums[endIndex];
                nums[endIndex] = temp;
            }
            startIndex++;
            while (nums[endIndex] == val) {
                endIndex--;
                if (endIndex <= startIndex) {
                    break;
                }
            }
        }
        return k;
    }
}